File: app/src/main/java/me/imzack/app/end/injector/component/PlanDetailComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.PlanDetailPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.PlanDetailActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = PlanDetailPresenterModule.class)
public interface PlanDetailComponent {

    void inject(PlanDetailActivity planDetailActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/PlanDetailComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.PlanDetailPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.PlanDetailActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(PlanDetailPresenterModule::class))
interface PlanDetailComponent {

    fun inject(planDetailActivity: PlanDetailActivity)
}
